kong/plugins/http-log/migrations/001_230_to_240.lua

```diff
@@ -0,0 +1,36 @@
+local operations = require "kong.db.migrations.operations.200_to_210"
+
+
+local function ws_migration_teardown(ops)
+  return function(connector)
+    return ops:fixup_plugin_config(connector, "http-log", function(config)
+      local updated = false
+      if type(config) == "table" then -- not required, but let's be defensive here
+        local headers = config.headers
+        if type(headers) == "table" then
+          for header_name, value_array in pairs(headers) do
+            if type(value_array) == "table" then
+              -- only update if it's still a table, so it is reentrant
+              headers[header_name] = value_array[1] or "empty header value"
```

Comment:
Shouldn't this value be an actual empty string?

```suggestion
              headers[header_name] = value_array[1] or ""
```

Reply:
no, in the old implementation one could set an empty array. In that case we now transform it to "empty header value" as a clear(?) indicator there was nothing there. Functionally the user should remove the header all together in that case. The alternative would be to simply drop the entry, but that would be too magical imo.

Reply:
btw empty headers are not valid